lib/system/threads.nim

```diff
@@ -642,12 +642,15 @@ when defined(windows):
 
 elif defined(linux):
   proc syscall(arg: clong): clong {.varargs, importc: "syscall", header: "<unistd.h>".}
-  var NR_gettid {.importc: "__NR_gettid", header: "<sys/syscall.h>".}: int
+  when defined(amd64):
+    const NR_gettid = cint(186)
+  else:
+    var NR_gettid {.importc: "__NR_gettid", header: "<sys/syscall.h>".}: cint
 
   proc getThreadId*(): int =
     ## get the ID of the currently running thread.
     if threadId == 0:
-      threadId = int(syscall(NR_gettid))
+      threadId = int(syscall(clong(NR_gettid)))
```

Comment:
Why do this type conversion here? Make NR_gettid of type ``clong`` (or ``int``).

Reply:
mainly because all other macro-derived constants (for example in posix.nim) are cint (used detect.nim to grab the value only to realize that this code can't use posix.nim anyway...